web-transport-wasm/src/session.rs

```diff
@@ -1,35 +1,34 @@
 use bytes::Bytes;
-use js_sys::Uint8Array;
-use wasm_bindgen::JsCast;
+use js_sys::{Reflect, Uint8Array};
 use wasm_bindgen_futures::JsFuture;
 use web_sys::{
     WebTransportBidirectionalStream, WebTransportCloseInfo, WebTransportReceiveStream,
     WebTransportSendStream,
 };
 
-use crate::{Reader, RecvStream, SendStream, WebError, Writer};
+use crate::{Reader, RecvStream, SendStream, SessionError, WebErrorExt, Writer};
 
 #[derive(Clone)]
 pub struct Session {
     inner: web_sys::WebTransport,
 }
 
 impl Session {
-    pub async fn new(url: &str) -> Result<Self, WebError> {
-        let inner = web_sys::WebTransport::new(url)?;
-        JsFuture::from(inner.ready()).await?;
+    pub async fn connect(url: &str) -> Result<Self, SessionError> {
```

Comment:
> Hey @kixelated - thanks for your work on this crate! Since you have this PR open would it be possible to add the ability to use `web_sys::WebTransport::new_with_options` https://rustwasm.github.io/wasm-bindgen/api/web_sys/struct.WebTransport.html#method.new_with_options here?
> 
> Since this PR is already introducing a breaking API change the function signature could be changed to this instead:
> 
> ```
>     pub async fn connect(url: &str, options: Option<web_sys::WebTransportOptions>) -> Result<Self, SessionError> 
> ```
> 
> This would make it much easier to use this crate when testing locally with self-signed certs. Happy to submit a separate PR if you prefer

Yep yep I have a branch with cert support that I can push/merge. I just got my WASM app working and didn't want to merge untested code until then.

Reply:
@rjobanp does #30 work for you?

Reply:
Specifically, I made a builder: https://github.com/kixelated/web-transport-rs/blob/e1e3e1a64e6c82393e5e8b062131a49d36636cd5/web-transport-wasm/src/session.rs#L113

Reply:
Yes that looks great to me - thank you!